aws-lc-rs/src/ec.rs

```diff
@@ -323,18 +319,40 @@ pub(crate) unsafe fn marshal_private_key_to_buffer(
     Ok(buffer)
 }
 
+pub(crate) unsafe fn unmarshal_der_to_private_key(
+    key_bytes: &[u8],
+    nid: i32,
+) -> Result<LcPtr<EVP_PKEY>, KeyRejected> {
+    let mut out = null_mut();
+    let evp_pkey = LcPtr::new(aws_lc::d2i_PrivateKey(
+        EVP_PKEY_EC,
+        &mut out,
+        &mut key_bytes.as_ptr(),
+        key_bytes
+            .len()
+            .try_into()
+            .map_err(|_| KeyRejected::too_large())?,
+    ))?;
+    validate_evp_key(&evp_pkey.as_const(), nid)?;
```

Comment:
I think this is redundant, [d2i_PrivateKey](https://github.com/aws/aws-lc/blob/main/crypto/evp_extra/p_ec_asn1.c#L264) ends up calling [EC_KEY_parse_private_key](https://github.com/aws/aws-lc/blob/main/crypto/evp_extra/p_ec_asn1.c#L158) which calls [EC_KEY_check_key](https://github.com/aws/aws-lc/blob/main/crypto/ec_extra/ec_asn1.c#L188).

Reply:
Removed in the next commit.